plan/logical_plan_builder.go

```diff
@@ -442,8 +442,21 @@ func (b *planBuilder) buildProjection(p LogicalPlan, fields []*ast.SelectField,
 				}
 			} else {
 				innerExpr := getInnerFromParentheses(field.Expr)
-				if _, ok := innerExpr.(*ast.ValueExpr); ok && innerExpr.Text() != "" {
-					colName = model.NewCIStr(innerExpr.Text())
+				if valueExpr, ok := innerExpr.(*ast.ValueExpr); ok {
+					switch valueExpr.Kind() {
+					case types.KindString:
+						// See #3686
+						colName = model.NewCIStr(valueExpr.GetString())
+					case types.KindNull:
+						// See #3686
+						colName = model.NewCIStr("NULL")
+					default:
```

Comment:
For KindNull: executor_test.go @ Line 288
For KindDefault: this is the original behavior. I think new tests are not needed.